### PR TITLE
API Changes: HTTP Documentation v2

### PR DESCRIPTION
**Pull request by amandawrcoelho**
Describe  some enums

Signed-off-by: Amanda Coelho <amada.winder.ribeiro.coelho@intel.com>

### CODE REVIEW

File: src/lib/comms/include/sol-http.h [hunk 1]
```diff
@@ -45,15 +45,49 @@ extern "C" {
...
+     * Performs a message loop-back test along the path to the
+     * target resource
+     */
     SOL_HTTP_METHOD_TRACE,
+    /**
+     * Used to update partial resources
+     */
     SOL_HTTP_METHOD_PATCH,
+    /**
+     * Invalid input method
```
Comment (bdilly) [inline src/lib/comms/include/sol-http.h]: not clear imo
I believe we could elaborate a bit more about it

**Comment by bdilly** [comment]
seems good overall

File: src/lib/comms/include/sol-http.h (hunk 1 above)
Comment (amandawrcoelho) [inline src/lib/comms/include/sol-http.h]: What do you about "client makes an HTTP request by using an HTTP method that does not comply with the HTTP specifications"?

Comment (bdilly) [inline src/lib/comms/include/sol-http.h]: yeah, it's better.
thanks

**Comment by amandawrcoelho** [comment]
@bdilly tks for the review! :D

**Pull request closed without merging**